packages/core/src/item.ts

```diff
@@ -296,22 +296,37 @@ export async function createVaultItem({
     });
 }
 
-const matchUsername = /username/i;
-const matchPassword = /password/i;
-const matchUrl = /url/i;
-const matchNote = /\n/;
-
 /** Guesses the most appropriate field type based on field name and value */
-export function guessFieldType({ name = "", value = "", masked }: any): FieldType {
-    return masked || name.match(matchPassword)
-        ? FieldType.Password
-        : name.match(matchUsername)
-        ? FieldType.Username
-        : name.match(matchUrl)
-        ? FieldType.Url
-        : value.match(matchNote)
-        ? FieldType.Note
-        : FieldType.Text;
+export function guessFieldType({
+    name,
+    value = "",
+    masked = false,
+}: {
+    name: string;
+    value?: string;
+    masked?: boolean;
+}): FieldType {
+    if (masked) {
+        return FieldType.Password;
+    }
+
+    let matchedTypeByName = Object.keys(FIELD_DEFS).filter((fieldType) =>
```

Comment:
Minor nitpick, by why not `const`?

Reply:
I was making some changes before, it should totally be `const` when not reassigned.